fix: redirect to user details page when session is valid

natalian98 opened the pull request:
This PR changes the login flow creation logic:
- Users who are already authenticated are redirected to the User Management page instead of initiating another flow.
- Users without a session or who have not completed 2FA when required proceed with the login flow as before.

closes https://github.com/canonical/identity-platform-login-ui/issues/768

natalian98 committed:
fix: redirect to user details page when session is valid

natalian98 committed:
refactor(ui): change settings page in error handler

nsklikas commented on pkg/kratos/handlers.go:
```diff
@@ -137,6 +144,13 @@ func (a *API) handleCreateFlow(w http.ResponseWriter, r *http.Request) {
 	if err != nil {
+		// Propagate the KratosErrorResponse so frontend can handle it
+		if kratosError, ok := parseGenericError(err); ok {
+			w.WriteHeader(http.StatusInternalServerError)
```
thought: perhaps a different error code should be used. It isn't a server error and it would also make the frontend logic simpler, as they can reason about the type of error based on the http status code.

natalian98 replied:
Switched to 400 to follow the status returned by kratos

nsklikas reviewed (dismissed):
lgtm

anusha-c18 committed:
feat: initiate login in a new context

natalian98 commented:
let's merge https://github.com/canonical/identity-platform-login-ui/pull/697 before this one

natalian98 committed:
fix: propagate aal2 to determine if cookies must be deleted

nsklikas reviewed (commented):
I got the following bugs:
1. If I don't complete 2fa, the flow is broken. Setup an account with 2fa and:
- Go to the log in page
- Enter username/password
- instead of completing 2fa, go back
You won't be able to log in as the filter cookie functionality is removed.

2. On first login, instead of being redirected back to the app I end up to the manage_details page.

natalian98 committed:
fix: return error on GetLoginFlow and delete session cookie in case of csrf violation

natalian98 committed:
fix: respect max_age
- propagate aal2 only for kratos login flow (flows without login challenge)
- remove kratos session cookie on identifier_first method

nsklikas reviewed (dismissed):
There is a minor issue, but realistically speaking it should not occur with the current deployment config. The rest of the comments are about personal preferences, so I am approving the PR and you can choose if you wish to keep them.

natalian98 committed:
refactor: check for session instead of aal2

wood-push-melon commented on pkg/kratos/handlers.go:
```diff
@@ -195,12 +213,26 @@ func (a *API) handleCreateFlowNewSession(r *http.Request, aal, returnTo, loginCh
 	return response, cookies, nil
 }
 
+func parseGenericError(err error) (*KratosErrorResponse, bool) {
```
(feel free to ignore, I might be pedantic and wrong):

We'd better follow the Go idiomatics to return `(value, error)` somehow if we can.
